src/librustc_resolve/diagnostics.rs

```diff
@@ -667,6 +669,10 @@ impl<'a> Resolver<'a> {
                     return;
                 }
 
+                if via_import && name_binding.is_possibly_imported_variant() {
```

Comment:
Could you add the "There is an assumption elsewhere that paths of variants are ..." explanation as a comment here?

Reply:
Sure, done